### PR TITLE
Add RoundIdentity layer to transform numeric values to integer ids

### LINKED ISSUE #1841: Implement a keras layer to round numeric values and truncate values with a max value.
For numeric features, we may want to convert the values to integer id by rounding and then make embedding with those ids.  Though we can implement the function by using bucketing with boundaries where 1 interval，it is tedious to set the boundaries and inefficient to transform if the max value is big.

Example：
inputs: [[1.2], [4.1], [6.7], [100.1]] 
max_value: 50
default_value: 50
the outputs: [[1], [4], [7], [50]]

### PR DESCRIPTION
**Pull request by workingloong**
Fix #1841

### CODE REVIEW

File: elasticdl_preprocessing/layers/round_identity.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import tensorflow as tf
+from tensorflow.python.ops.ragged import ragged_functional_ops, ragged_tensor
+
+
+class RoundIdentity(tf.keras.layers.Layer):
+    """Implements numeric feature rounding with a max value.
```
Comment (brightcoder01) [inline elasticdl_preprocessing/layers/round_identity.py]: I think this layer is casting a numeric feature into a categorical feature.

File: elasticdl_preprocessing/layers/round_identity.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        inp = np.asarray([[1.2], [1.6], [0.2], [3.1], [4.9]])
+        layer(inputs)
+        [[1], [2], [0], [3], [5]]
+    ```
+
+    Arguments:
+        num_buckets: Range of inputs and outputs is `[0, num_buckets)`.
+        **kwargs: Keyword arguments to construct a layer.
+
+    Input shape: A numeric tensor of shape
```
Comment (brightcoder01) [inline elasticdl_preprocessing/layers/round_identity.py]: Tensor and SparseTensor

File: elasticdl_preprocessing/layers/round_identity.py (hunk 1 above)
Comment (workingloong) [reply elasticdl_preprocessing/layers/round_identity.py]: How do you define the categorical feature? I think the categorical feature is the type fo source feature. The layer is a tool to process the numeric feature although the outputs are integer ids.

File: elasticdl_preprocessing/layers/round_identity.py (hunk 2 above)
Comment (workingloong) [reply elasticdl_preprocessing/layers/round_identity.py]: Done

File: elasticdl_preprocessing/layers/round_identity.py (hunk 1 above)
Comment (brightcoder01) [reply elasticdl_preprocessing/layers/round_identity.py]: Maybe: cast a numeric feature into a d discretized/categorical value.
IMO, categorical value is discretized and enumerable. After executing this layer, the value become enumerable: [0, 1, 2, 3, ..., MAX_VALUE]

Comment (workingloong) [reply elasticdl_preprocessing/layers/round_identity.py]: Cast a numeric feature into a discrete integer value?

Comment (brightcoder01) [reply elasticdl_preprocessing/layers/round_identity.py]: discretized integer value?

**Review by brightcoder01 — APPROVED** [review]

File: elasticdl_preprocessing/layers/round_identity.py (hunk 1 above)
Comment (workingloong) [reply elasticdl_preprocessing/layers/round_identity.py]: "discrete" is an adj and the code spell checker in VScode hints that "discretized" is an unknown word.

**Pull request merged**